dash/_utils.py

```diff
@@ -1,3 +1,24 @@
+import functools
+import hashlib
+import base64
+import pkgutil
+
+
+def memoize(func):
+    results = {}
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        key = hash((args, frozenset(kwargs)))
```

Comment:
should this be `hash((args, frozenset(kwargs.keys()), frozenset(kwargs.values()))`?

`frozenset(kwargs)` gets a set of the keys, so `function(1, 2, a=1)` and `function(1, 2, a=2)` would produce the same hash.

Reply:
Good catch, the function it wraps doesn't have kwargs so it didn't impact it. I think frozenset(kwargs.items()) so the values and keys are grouped together in a tuple?